tests/rptest/tests/wasm_basic_test.py

```diff
@@ -0,0 +1,122 @@
+                                            extra_rp_conf=extra_rp_conf)
+
+        self.test_file_path = os.path.join(self.resource_dir(),
+                                           "wasm-identity-transform.js")
+
+        if not os.path.exists(self.test_file_path):
+            raise Exception(
+                f"This test expects {self.test_file_path} to be in the resources dir"
+            )
+
+        self.rpk_tool = RpkTool(self.redpanda)
+        self.kafka_tools = KafkaCliTools(self.redpanda)
+
+    def _read_records(self, tps, n):
+        # Attempt to read all 1024 records
+        records = []
+        consumer = None
+        try:
+            consumer = KafkaConsumer(client_id=uuid.uuid4(),
+                                     bootstrap_servers=self.redpanda.brokers(),
+                                     request_timeout_ms=1000,
+                                     enable_auto_commit=False,
+                                     auto_offset_reset="earliest")
+        except ValueError as e:
+            self.redpanda.logger.error(
+                f"Failed to create KafkaConsumer: {str(e)})")
+            return []
+
+        consumer.assign(tps)
+        empty_iterations = 10
+        while len(records) < n and empty_iterations > 0:
+            r = consumer.poll(timeout_ms=100, max_records=1024)
+            for data in r.values():
+                records += data
+            # If we've looped 'empty_iterations' times without data, exit
+            if len(r) == 0:
+                empty_iterations -= 1
+        consumer.close()
+        return records
+
+    # For now we are deploying a static pre generated coprocessor, it:
+    # Subscribes to a topic called 'myInputTopic' and performs no transform,
+    # just returns same records on a new topic 'myOutputTopic'
+    @cluster(num_nodes=3)
+    def wasm_basic_test(self):
+        # Produce all data
+        def produce_until_partitions():
+            self.kafka_tools.produce(self.topic, 1024, 1024)
+            storage = self.redpanda.storage()
+            return len(list(storage.partitions("kafka", self.topic))) == 9
+
+        wait_until(lambda: produce_until_partitions(),
+                   timeout_sec=10,
+                   backoff_sec=2,
+                   err_msg="Expected partition did not materialize")
+
+        # Deploy coprocessor
+        self.rpk_tool.wasm_deploy(self.test_file_path, "ducktape")
+
+        # Wait until some data hits disk
+        def materialized_topics_exist():
+            storage = self.redpanda.storage()
+            partitions = storage.partitions("kafka",
+                                            WasmBasicTest.materialized_topic)
+            return len(list(partitions)) == 9
+
+        wait_until(lambda: materialized_topics_exist(),
+                   timeout_sec=10,
+                   backoff_sec=2,
+                   err_msg="Materialized topics haven't been created")
```

Comment:
Nice! Using the `self.redpanda.storage()` service is really powerful for writing tests that inspect assumptions about storage. I think this test is ok, but the goal was testing storage-level partitions, it should be named something more specific than `basic_test`--something like `ensure_materialized_partitions` and then remove the step of reading the records at the end.

If the goal was creating a simple producer/consumer test then I think that you should remove this `wait_until` and the one above for producing because they introduce synchronization points in this functional test that won't be present in a real world scenario.